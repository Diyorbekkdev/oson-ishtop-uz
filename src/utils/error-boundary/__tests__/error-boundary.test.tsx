import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { ErrorBoundary } from "..";

const ThrowError: React.FC = () => {
	throw new Error("Test error");
};

describe("ErrorBoundary Component", () => {
	it("should render fallback UI when an error is thrown", () => {
		const { asFragment } = render(
			<ErrorBoundary data-testid="error-component">
				<ThrowError />
			</ErrorBoundary>,
		);
		expect(screen.getByTestId("error-component")).toBeInTheDocument();
		expect(screen.getByText("Something is wrong")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
