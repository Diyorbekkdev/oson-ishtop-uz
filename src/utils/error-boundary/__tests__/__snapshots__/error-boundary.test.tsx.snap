// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBoundary Component > should render fallback UI when an error is thrown 1`] = `
<DocumentFragment>
  <h1
    data-testid="error-component"
  >
    Something is wrong
  </h1>
</DocumentFragment>
`;
